mpi4jax/_src/tokenizer.py

```diff
@@ -0,0 +1,154 @@
+import jax
+from jax._src.util import safe_map
```

Comment:
I'm a bit worried about stability here. We've seen many breakages due to JAX shuffling around stuff. Can we implement this so it only depends on the public API? Maybe vendor `safe_map`?

Reply:
Let me just copy and paste the function haha.

Reply:
Could you add a comment that this is taken from JAX and include a link to the license? Wouldn't want Google lawyers to come knocking :wink: